### PR TITLE
test(observability): cover logger service with unit tests

### PR DESCRIPTION
**Pull request by forbesus**
ref #429

### CODE REVIEW

**Comment by forbesus** [comment]
hello @ygrishajev 
i created unit test
would you check it?

File: packages/logging/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    preset: "ts-jest",
+    testEnvironment: "node",
+    testMatch: ["**/test/**/*.test.ts"],
```
Comment (ygrishajev) [inline packages/logging/jest.config.js]: let's make it like in other apps/packages pls
```suggestion
    testMatch: ["**/test/**/*.spec.ts"],
```

File: packages/logging/package.json [hunk 2]
```diff
@@ -6,18 +6,22 @@
...
+    "@types/jest": "^29.5.14",
     "dotenv": "^16.4.5",
     "http-errors": "^2.0.0",
+    "jest": "^29.7.0",
     "pino": "^9.5.0",
     "pino-cloud-logging": "^1.0.6",
     "pino-fluentd": "^0.2.4",
     "pino-pretty": "^11.3.0",
     "stream": "^0.0.3",
+    "ts-jest": "^29.2.5",
```
Comment (ygrishajev) [inline packages/logging/package.json]: could you please make sure the versions of new packages here are the same as in other projects? you may check against `apps/api`

File: packages/logging/test/LoggerService.test.ts
Comment (ygrishajev) [inline packages/logging/test/LoggerService.test.ts:1]: please move this file to `packages/logging/src/servicies/logger/logger.service.spec.ts`

File: packages/logging/test/LoggerService.test.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  afterEach(() => {
+    config.STD_OUT_LOG_FORMAT = defaultLogFormat;
+    config.FLUENTD_TAG = defaultFluentdTag;
+    config.FLUENTD_HOST = defaultFluentdHost;
+    config.FLUENTD_PORT = defaultFluentdPort;
+    jest.clearAllMocks();
+  });
+
+  describe("initPino", () => {
```
Comment (ygrishajev) [inline packages/logging/test/LoggerService.test.ts]:
```suggestion
  describe("constructor", () => {
```

File: packages/logging/test/LoggerService.test.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      expect(loggable).toEqual({
+        status: 404,
+        message: "Not found",
+        stack: "stack trace",
+        data: { key: "value" },
+        originalError: "stack trace"
+      });
+    });
+
+    it("should return status, message, stack, and data for HttpError", () => {
```
Comment (ygrishajev) [inline packages/logging/test/LoggerService.test.ts]: seems to be a duplicate of the above

File: packages/logging/test/LoggerService.test.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+      config.FLUENTD_HOST = "";
+      config.FLUENTD_PORT = 0;
+      config.FLUENTD_TAG = "";
+
+      new LoggerService();
+      expect(pinoFluentd).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("Logging methods", () => {
```
Comment (ygrishajev) [inline packages/logging/test/LoggerService.test.ts]: I suggest to adjust like so:

```typescript
const methods: (keyof Logger)[] = ["info", "error", "warn", "debug"];
  describe.each(methods)("prototype.%s", method => {
    const logMessage = "Test message";

    it(`should call pino.${method} on info method`, () => {
      loggerService[method](logMessage);
      expect(mockLogger[method]).toHaveBeenCalledWith(logMessage);
    });
  });
```

for this Logger would need to implement common interface, which can be done like so:

```typescript
import pino, { Logger as PinoLogger } from "pino";
export type Logger = Pick<PinoLogger, "info" | "error" | "warn" | "debug">;

export class LoggerService implements Logger {
```

**Review by ygrishajev — COMMENTED** [review]
hey, generally looks good. Adding some comments.

also commit should have type `test` and scope `observability`:

`test(observability): cover logger service with unit tests`

File: packages/logging/test/LoggerService.test.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+      expect(mockLogger.warn).toHaveBeenCalledWith(logMessage);
+    });
+
+    it("should call pino.debug on debug method", () => {
+      loggerService.debug(logMessage);
+      expect(mockLogger.debug).toHaveBeenCalledWith(logMessage);
+    });
+  });
+
+  describe("toLoggableInput", () => {
```
Comment (ygrishajev) [inline packages/logging/test/LoggerService.test.ts]:
```suggestion
  describe("prototype.toLoggableInput", () => {
```

File: packages/logging/src/servicies/logger/logger.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+  afterEach(() => {
+    config.STD_OUT_LOG_FORMAT = defaultLogFormat;
+    config.FLUENTD_TAG = defaultFluentdTag;
+    config.FLUENTD_HOST = defaultFluentdHost;
+    config.FLUENTD_PORT = defaultFluentdPort;
+    jest.clearAllMocks();
+  });
+
+  describe("constructor", () => {
```
Comment (ygrishajev) [inline packages/logging/src/servicies/logger/logger.service.spec.ts]: ouch, I've just noticed there's the actual method `initPino`. So the describe you had before actually makes more sense! apologies!
```suggestion
  describe("prototype.initPino", () => {
```

If you don't want to change it, I'll just do it myself later :)

**Review by ygrishajev — APPROVED** [review]

**Comment by forbesus** [comment]
@ygrishajev all things are changed

**Comment by ygrishajev** [comment]
thanks!

**Pull request merged**